cmd/artifactPrepareVersion_test.go

```diff
@@ -0,0 +1,590 @@
+
+		tt := []struct {
+			repo          gitRepositoryMock
+			sshAgentAuth  func(string) (*ssh.PublicKeysCallback, error)
+			expectedError string
+		}{
+			{repo: gitRepositoryMock{remote: remote, deleteRemoteError: []string{"delete error"}}, sshAgentAuth: sshSuccess, expectedError: "failed to update remote origin - remove: delete error"},
+			{repo: gitRepositoryMock{remote: remote, createRemoteError: []string{"update error"}}, sshAgentAuth: sshSuccess, expectedError: "failed to update remote origin - create: update error"},
+			{repo: gitRepositoryMock{remote: remote}, sshAgentAuth: sshFailure, expectedError: "failed to retrieve ssh authentication: ssh error"},
+			{repo: gitRepositoryMock{remote: remote, deleteRemoteError: []string{"", "delete error"}}, sshAgentAuth: sshSuccess, expectedError: "failed to restore remote origin - remove: delete error"},
+			{repo: gitRepositoryMock{remote: remote, createRemoteError: []string{"", "update error"}}, sshAgentAuth: sshSuccess, expectedError: "failed to restore remote origin - create: update error"},
+		}
+
+		originalSSHAgentAuth := sshAgentAuth
+
+		for _, test := range tt {
+			sshAgentAuth = test.sshAgentAuth
+			commitID, err := pushChanges(&config, newVersion, &test.repo, &worktree, testTime)
+			sshAgentAuth = originalSSHAgentAuth
+
+			assert.Equal(t, "428ecf70bc22df0ba3dcf194b5ce53e769abab07", commitID)
+			assert.EqualError(t, err, test.expectedError)
+		}
+	})
+
+	t.Run("error - push", func(t *testing.T) {
+		config := artifactPrepareVersionOptions{Username: "testUser", Password: "****"}
+		repo := gitRepositoryMock{remote: remote, pushError: "push error"}
+		worktree := gitWorktreeMock{commitHash: plumbing.ComputeHash(plumbing.CommitObject, []byte{1, 2, 3})}
+
+		commitID, err := pushChanges(&config, newVersion, &repo, &worktree, testTime)
+		assert.Equal(t, "428ecf70bc22df0ba3dcf194b5ce53e769abab07", commitID)
+		assert.EqualError(t, err, "push error")
+	})
+}
+
+func TestTemplateCompatibility(t *testing.T) {
+	tt := []struct {
+		groovy         string
+		versioningType string
+		timestamp      bool
+		commitID       bool
+	}{
+		{groovy: `${version}`, versioningType: "library", timestamp: false, commitID: false},
+		{groovy: `${version}-${timestamp}`, versioningType: "cloud", timestamp: true, commitID: false},
+		{groovy: `${version}-${timestamp}${commitId?"_"+commitId:""`, versioningType: "cloud", timestamp: true, commitID: true},
+	}
+
+	for _, test := range tt {
+		versioningType, timestamp, commitID := templateCompatibility(test.groovy)
+		assert.Equal(t, test.versioningType, versioningType)
+		assert.Equal(t, test.timestamp, timestamp)
+		assert.Equal(t, test.commitID, commitID)
+	}
+}
+
+func TestConvertHTTPToSSHURL(t *testing.T) {
+	tt := []struct {
+		httpURL  string
+		expected string
+	}{
+		{httpURL: "https://my.test.server/owner/repo.git", expected: "git@my.test.server:owner/repo.git"},
```

Comment:
Minor: There could be more cases, for example what if the URL already is in ssh format?